aten/src/ATen/native/Histogram.cpp

```diff
@@ -0,0 +1,133 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/NativeFunctions.h>
+
+#include <ATen/native/Histogram.h>
+
+#include <tuple>
+
+namespace at { namespace native {
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+DEFINE_DISPATCH(histogram_stub);
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+DEFINE_DISPATCH(histogram_linear_stub);
+
+namespace {
+
+inline void histogram_pre_check(const Tensor& input, const Tensor& bins, const c10::optional<Tensor>& weight,
+    bool density, const Tensor& hist, const Tensor& bin_edges) {
+  TORCH_CHECK(bins.device() == input.device(), "bins and input value tensors should have same device type, ",
```

Comment:
Checks that all tensors are on the same device are generated automatically now. This is a recent change, but we can assume all tensors are on the same device and don't need to check redundantly